src/main/java/org/killbill/billing/client/KillBillHttpClient.java

```diff
@@ -117,12 +118,21 @@ public KillBillHttpClient(final String kbServerUrl, final String username, final
         this.apiKey = apiKey;
         this.apiSecret = apiSecret;
 
-
         final AsyncHttpClientConfig.Builder cfg = new AsyncHttpClientConfig.Builder();
 
+        if(requestTimeout != null)
+        {
+            cfg.setRequestTimeout(requestTimeout);
+            requestTimeoutSec = (int) Math.ceil((double) requestTimeout / 1000);
```

Comment:
Oh, I see, you're right. I didn't realize we had public methods where you could override the timeout.

In that case, could you use `TimeUnit` to simplify this? Something like `TimeUnit.MILLISECONDS.toSeconds(requestTimeout)`?

Note that it will round down (so the future will expire before the request in some cases), but the window is less than a second.